lib/ansible/plugins/action/aireos.py

```diff
@@ -38,42 +38,62 @@ def run(self, tmp=None, task_vars=None):
 
         module_name = self._task.action.split('.')[-1]
         self._config_module = True if module_name == 'aireos_config' else False
-
-        if self._play_context.connection != 'local':
+        persistent_connection = self._play_context.connection.split('.')[-1]
+        warnings = []
+
+        if persistent_connection == 'network_cli':
+            provider = self._task.args.get('provider', {})
+            if any(provider.values()):
+                display.warning('provider is unnecessary when using network_cli and will be ignored')
+                del self._task.args['provider']
+        elif self._play_context.connection == 'local':
+            provider = load_provider(aireos_provider_spec, self._task.args)
+            pc = copy.deepcopy(self._play_context)
+            pc.connection = 'ansible.netcommon.network_cli'
```

Comment:
How can we hardcode this here? This means that ansible.netcommon namespace is now locked into our codebase.

Reply:
Yes, the `ansible.netcommon` is locked and will be used to distribute all the common network code.